src/adapter/src/coord.rs

```diff
@@ -2858,6 +2858,37 @@ pub fn serve(
             .expect("missing EpochMillisseconds timestamp")
             .clone();
 
+        // Choose a time at which to boot. This is the time at which we will run
+        // internal migrations.
+        //
+        // This time is usually the current system time, but with protection
+        // against backwards time jumps, even across restarts.
+        let boot_ts = {
+            // WIP: @joe we have a chicken-and-egg problem here. Ideally we'd
+            // only use monotonic_now when we know we're using the Catalog
+            // oracle, and when using the Postgres oracle we'd only advance the
+            // timestamp by one. But we can only know that once we opened the
+            // Catalog.
```

Comment:
We have the same issue with `catalog_kind` and `persist_txn`. The solution is to store a copy of the system variable in the config collection, which we can read before opening the catalog.

Reply:
I can point you in the direction of where that code is if you want.